catalyst_voices/packages/catalyst_voices_models/lib/src/workspace/workspace_proposal_segment_step.dart

```diff
@@ -1,21 +1,20 @@
 import 'package:equatable/equatable.dart';
-import 'package:flutter_quill/flutter_quill.dart';
 
 class WorkspaceProposalSegmentStep extends Equatable {
   final int id;
   final String title;
   final String? description;
-  final Document? document;
+  final List<dynamic>? jsonData;
```

Comment:
maybe `Map<String, dynamic>` would be better

Reply:
Json for quill's `Document.fromJson` is in the List format

<img width="329" alt="Screenshot 2024-09-30 at 14 43 36" src="https://github.com/user-attachments/assets/9d5ae05c-df98-4854-a2a0-963a7a8b8a13">

Reply:
Usually passing raw jsons isn't recommended. There should be a data structure in models package which holds this json, even as a string, then on catalyst_voices package there should be an extension which converts this data structure to what you need. And here in the WorkspaceProposalSegmentStep you should pass this data structure.

Reply:
Can even be extension type instead of a class, here's an example:

```dart
extension type const Coin(int value) {
```